worker/worker-execution/score-worker-execution-impl/src/main/java/io/cloudslang/worker/execution/services/ExecutionServiceImpl.java

```diff
@@ -382,11 +408,25 @@ private void failFlowIfSplitStepFailed(Execution execution) throws InterruptedEx
             String branchId = splitUuid + ":" + branchIndexInSplitStep;
             to.getSystemContext().setBranchId(branchId);
             dispatchBranchStartEvent(executionId, splitUuid, branchId, currStep);
+            addCheckoutLaneFlagToContext(branchId, to.getSystemContext(), count);
             newExecutions.add(to);
         }
         return newExecutions;
     }
 
+    private void addCheckoutLaneFlagToContext(String branchId, SystemContext systemContext, int branchNumber) {
+        if ("NON_BLOCKING".equals(systemContext.get("STEP_TYPE"))) {
```

Comment:
seems to introduce some slowness, two if and one nested if.
I would prefer to use a single if
```
Integer parallelismLevel = ...
if (parallel && correctLane) {
     systemContext.put(BRANCH_ID_TO_CHECK_OUT_LICENSE, branchId);
}
```